### PR TITLE
fix(apiserver): Check nil for ExternalAddress

### PR DESCRIPTION
**Pull request by ijumps**
**What type of PR is this?**

/kind bug

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**: #72404

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: staging/src/k8s.io/apiserver/pkg/server/options/serving.go [hunk 1]
```diff
@@ -108,7 +108,7 @@ func NewSecureServingOptions() *SecureServingOptions {
 }
 
 func (s *SecureServingOptions) DefaultExternalAddress() (net.IP, error) {
-	if !s.ExternalAddress.IsUnspecified() {
+	if s.ExternalAddress != nil && !s.ExternalAddress.IsUnspecified() {
```
Comment (yue9944882) [inline staging/src/k8s.io/apiserver/pkg/server/options/serving.go]: didnt see an issue, whats ur binding address?

[https://github.com/kubernetes/kubernetes/blob/7c112762b0df87f3a1772f3bbede7fe9185fa249/staging/src/k8s.io/apimachinery/pkg/util/net/interface.go#L408](https://github.com/kubernetes/kubernetes/blob/7c112762b0df87f3a1772f3bbede7fe9185fa249/staging/src/k8s.io/apimachinery/pkg/util/net/interface.go#L408)

Comment (ijumps) [reply staging/src/k8s.io/apiserver/pkg/server/options/serving.go]: if `ExternalAddress` is nil, `DefaultExternalAddress` will just return nil, not goto `ChooseBindAddress` func. 

you can find `How to reproduce it` in https://github.com/kubernetes/kubernetes/issues/72404

**Pull request closed without merging**